src/Tribe/I18n.php

```diff
@@ -0,0 +1,217 @@
+		$domains = wp_parse_args(
+			$domains,
+			[
+				// The `default` domain doesn't need file path.
+				'default'             => true,
+				'the-events-calendar' => $this->tec->plugin_dir . 'lang/',
+			]
+		);
+
+		return $this->get_i18n_strings_for_domains( $strings, $languages, $domains );
+	}
+
+	/**
+	 * Get all possible translations for a String based on the given Languages and Domains
+	 *
+	 * WARNING: This function is slow because it deals with files, so don't overuse it!
+	 * Differently from the `get_i18n_strings` method this will not use any domain that's not specified.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $strings   An array of strings (required).
+	 * @param array $languages Which l10n to fetch the string (required).
+	 * @param array $domains   Possible domains to re-load.
+	 *
+	 * @return array<string,array|string> A multi level array with the possible translations for the given strings.
+	 *
+	 * @todo Include support for the `load_theme_textdomain` + `load_muplugin_textdomain`
+	 */
+	public function get_i18n_strings_for_domains( $strings, $languages, $domains = array( 'default' ) ) {
+		sort( $languages );
+		$strings_buffer = [ $strings ];
+
+		foreach ( $languages as $language ) {
+			// Override the current locale w/ the one we need to compile the translations.
+			$language_strings = $this->with_locale(
+				$language,
+				[ $this, 'compile_translations' ],
+				[ $strings, $domains ]
+			);
+			$strings_buffer[] = $language_strings;
+		}
+
+		$strings = count( $strings_buffer ) > 1
+			? array_merge_recursive( ... $strings_buffer )
+			: reset( $strings_buffer );
+
+		// Prevent empty strings and duplicates.
+		foreach ( $strings as $key => $value ) {
+			$strings[ $key ] = array_filter(
+				array_unique(
+					array_map( 'sanitize_title_with_dashes', (array) $value )
+				)
+			);
+		}
+
+		return $strings;
+	}
+
+	/**
+	 * Executes a callback ensuring the `current_locale` will be set to the specified language code.
+	 *
+	 * The method will backup and detach the functions and methods currently filtering the `locale` filter to execute
+	 * the callback in isolation and restore the filters after that.
+	 * The main purpose of this method is to avoid a rat race against plugins and themes that will filter the locale
+	 * by attaching the filtering method or function at `PHP_INT_MAX`.
+	 *
+	 * @since TBD
+	 *
+	 * @param string   $locale The locale to set for the execution of the callback.
+	 * @param callable $do     The callable to execute in the context of a specific locale.
+	 * @param array    $args   A set of arguments that will be passed to the callback.
+	 *
+	 * @return mixed The callback return value, if any.
+	 */
+	protected function with_locale( $locale, callable $do, array $args = [] ) {
+		global $wp_filter;
+		$locale_filters_backup = isset( $wp_filter['locale'] ) ? $wp_filter['locale'] : [];
+		$wp_filter['locale']   = $locale_filters_backup instanceof \WP_Hook ? new \WP_Hook() : [];
+
```

Comment:
:no_entry_sign: Overriding WordPress globals is prohibited. Found assignment to $wp_filter

<sub>hash: 482c448a2fe8cace88230675d4661b07</sub>

Reply:
This requires some explanation.

The filtering method I'm trying to avoid is hooked at `PHP_INT_MAX`.  

This leaves the following possiblities that I can see:

1. Hook at `PHP_INT_MAX` after that filter. This would not unhook and re-hook, but it would rely upon this code, or some other code, running as late as possible. This code runs at `init` of Main, though.
2. Remove that specific filtering method. This would mean adding, in this general purpose code, knowledge and information about particular code (WPML, in this instance). This would require maintenance and review over time.
3. Do 2, but filter what we want to unhook and re-hook. This removes particular knowledge from this general code, but does not remove the maintenance and update burden.
4. Do what I did and ensure we'll be, in the context of this call, the only ones acting on this filter.

I'm really open to alternatives.